games-action/dave_gnukem/patches/dave_gnukem-1.0.patchset

```diff
@@ -0,0 +1,180 @@
+ using namespace std;
+ 
+diff --git a/src/main.cpp b/src/main.cpp
+index 183b95a..a7ef3b9 100644
+--- a/src/main.cpp
++++ b/src/main.cpp
+@@ -51,7 +51,7 @@ Copyright (C) 1995-2018 David Joffe
+ #endif
+ #endif
+ 
+-#include <SDL_mixer.h>//For background music stuff
++#include <SDL/SDL_mixer.h>//For background music stuff
+ 
+ #include <map>
+ #include <string>
+diff --git a/src/sdl/djgraph.cpp b/src/sdl/djgraph.cpp
+index ea61fca..f3a7275 100644
+--- a/src/sdl/djgraph.cpp
++++ b/src/sdl/djgraph.cpp
+@@ -15,7 +15,7 @@ Copyright (C) 1997-2018 David Joffe
+ 
+ #include "../djgraph.h"
+ #include "../sys_log.h"
+-#include "SDL.h"
++#include <SDL/SDL.h>
+ 
+ #include <string.h>
+ 
+diff --git a/src/sdl/djsound.cpp b/src/sdl/djsound.cpp
+index 5a3e2c9..8ebb875 100644
+--- a/src/sdl/djsound.cpp
++++ b/src/sdl/djsound.cpp
+@@ -7,10 +7,10 @@ Copyright (C) 1999-2018 David Joffe and Kent Mein
+ #include "../djsound.h"
+ #include "../djstring.h"
+ #include "../djlog.h"
+-#include <SDL_audio.h>
+-#include <SDL_error.h>
++#include <SDL/SDL_audio.h>
++#include <SDL/SDL_error.h>
+ #ifndef NOSOUND
+-#include <SDL_mixer.h>
++#include <SDL/SDL_mixer.h>
+ #endif
+ 
+ #ifdef __APPLE__
+diff --git a/src/sdl/djtime.cpp b/src/sdl/djtime.cpp
+index 2fa173d..355fe90 100644
+--- a/src/sdl/djtime.cpp
++++ b/src/sdl/djtime.cpp
+@@ -5,7 +5,7 @@ Copyright (C) 1999-2018 David Joffe
+ */
+ 
+ #include "../djtime.h"
+-#include "SDL.h"
++#include <SDL/SDL.h>
+ #ifdef WIN32
+ #include <windows.h>
+ #include <mmsystem.h>
+-- 
+2.19.0
+
+
+From 904ff4251d0ce803c71ff1ff42b6827bcb124ab6 Mon Sep 17 00:00:00 2001
+From: Abdur-Raheem Idowu <abdurraheemidowu@gmail.com>
+Date: Sat, 28 Dec 2019 10:45:15 +0000
+Subject: Allows lauch from deskbar
+
+
+diff --git a/src/main.cpp b/src/main.cpp
+index a7ef3b9..7742817 100644
+--- a/src/main.cpp
++++ b/src/main.cpp
+@@ -38,6 +38,11 @@ Copyright (C) 1995-2018 David Joffe
+ #include "sys_log.h"
+ #include "sys_error.h"
+ 
++#ifdef __HAIKU__
++#include <libgen.h>
++#include <unistd.h>
++#endif
++
+ #ifdef __APPLE__
+ #include <mach-o/dyld.h>//_NSGetExecutablePath
+ #include <sys/stat.h>//For djFolderExists stuff
+@@ -114,6 +119,11 @@ CMenu mainMenu ( "main.cpp:mainMenu" );
+ // This is the 'main' function. The big cheese.
+ int main ( int argc, char** argv )
+ {
++	
++#ifdef __HAIKU__
++	chdir(dirname(argv[0]));
++#endif
++
+ 	// Check commandline args
+ 	bool bfullscreen = false;
+ 	bool b640 = false;
+-- 
+2.19.0
+
+
+From 9a4f271e7082e3fd69fdf03d017c8bb9a49b1e39 Mon Sep 17 00:00:00 2001
+From: Abdur-Raheem Idowu <abdurraheemidowu@gmail.com>
+Date: Sat, 28 Dec 2019 10:46:47 +0000
+Subject: Corrected settings directory
+
+
+diff --git a/src/djstring.cpp b/src/djstring.cpp
+index 76237f9..30b0643 100644
+--- a/src/djstring.cpp
++++ b/src/djstring.cpp
+@@ -284,6 +284,9 @@ std::string djGetFolderUserSettings()
+ #ifdef WIN32
+ 	std::string s = getenv("APPDATA");
+ 	djAppendPathS(s, "DaveGnukem");
++#elif __HAIKU__
++	std::string s = getenv("HOME");
```

Comment:
use find_directory to find the settings dir. It may move in the future, that's why we have an API for it.